components/gmConnection.js

```diff
@@ -48,6 +48,9 @@ gmConnection.prototype = {
   },
 
   send: function(aUrl, aData) {
+    // Clear data
+    this._data = "";
```

Comment:
As you can see in lines 69-71 there is appending the stream data to the _data variable. If you're using the send function on the same instance of gmConnection twice then data from the second request is appended to the data from the first request which is bad.